packages/plugin-page-view-tracking/src/page-view-tracking.ts

```diff
@@ -0,0 +1,75 @@
+import { IAmplitudeBrowser } from '@amplitude/analytics-browser';
+import {
+  BaseEvent,
+  BeforePlugin,
+  BrowserClient,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import { PluginCampaignTracker } from './plugin-campaign-tracker';
+import { PageTrackingBrowserOptions } from './typings/page-view-tracking';
+
+export class PageViewTrackingPlugin implements BeforePlugin {
+  name = 'page-view-tracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(
+    private readonly instance: BrowserClient | IAmplitudeBrowser,
```

Comment:
Do we need `IAmplitudeBrowser`? I feel like customers do not have access to this one anyway
```suggestion
    private readonly instance: BrowserClient | IAmplitudeBrowser,
```

Reply:
Right. Because we need to add plugins in the `analytics-ma-browser` init function and pass browser options to plugins, then it would need `IAmplitudeBrowser`. 

```js
async init(apiKey, deviceId, options) {
  this.add(new AttributionPlugin(this, options?.attribution)); // the type of "this" is IAmplitudeBrowser
  ...
}
```

Reply:
hmm can we commit this later when we do it? wanted to think about this since it's a little confusing for customers to see a foreign interface plus promise interface is different

Reply:
Sure, I will remove this part from this PR.